rclcpp/doc/proposed_node_parameter_callbacks.md

```diff
@@ -0,0 +1,21 @@
+# Proposed node parameters callback Design
+
+## Introduction:
+
+ The original requirement came in **gazeb_ros_pkgs** for setting individual wheel slip params based on global wheel slip value [link to original issue](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1365). The main idea was to set some parameters using ```set_parameter``` API once the a given param was set successfully. The requirement let to the discussion of supporting registered callbacks once the parameters are set and have been requested by users before. Some related discussion here [#609](https://github.com/ros2/rclcpp/issues/609) [#1789](https://github.com/ros2/rclcpp/pull/1789).
```

Comment:
style nitpick: please use one sentence per line to help keep future diffs easier to read (see our [style guide](https://docs.ros.org/en/rolling/Contributing/Code-Style-Language-Versions.html#markdown-restructured-text-docblocks))